modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/SystemTimeBasedScheduler.java

```diff
@@ -91,7 +91,7 @@ public void run() {
                         scheduledExecutorService.schedule(eventCaller, toNotifyTime - currentTime, TimeUnit
                                 .MILLISECONDS);
                     } else {
-                        synchronized (toNotifyQueue) {
+                        synchronized (this) {
```

Comment:
Does doing synchronized (this) in inner class will not stop its outer class. 
So I think this is wrong.